.github/labeler.yml

```diff
@@ -0,0 +1,12 @@
+# if a label exists in nixpkgs then use that instead of making a new one
+
+"8.has: changelog":
+  - /doc/manual/src/release-notes/*
+
+"8.has: documentation":
```

Comment:
@Artturin Thank you, I think this is the exact right way to improve discoverability of pull requests.

I also think it's generally good to converge on a labeling scheme with Nixpkgs (but not at all costs), and that top comment is valuable.

I don't feel qualified to decide on which labels to use exactly and how to organise them, and in any case would like to know what @edolstra thinks about this topic.

A few questions for me to get a better idea:

Why are the Nixpkgs labels structured this way (number.verb: noun)? Why isn't it just "with documentation", for example? (It doesn't seem right to say "has documentation" because often changes just update docs instead of adding it.)

How do we deal with semantically redundant labels? If we changed "documentation" to "8.has: documentation", this will break existing queries, and we'd have to update contributing guides for example.

In general I think it would be best to add a note to the contributing section in the manual when introducing such a workflow, such that contributors (and maintainers) know what all the colorful, weirdly named labels mean and how to best make use of them.

So, before going ahead I'd like to see more explicit deliberation and ensure it will also be visible for others, i.e., not just those participating in this pull request.

What do you think?

Reply:
> Why are the Nixpkgs labels structured this way (number.verb: noun)? Why isn't it just "with documentation", for example?

it's for sorting
nixpkgs has [171 labels](https://github.com/NixOS/nixpkgs/labels)
nix has [37 labels](https://github.com/NixOS/nix/labels) so the numbering might not be necessary

![image](https://user-images.githubusercontent.com/56650223/207114152-93782817-635a-41b5-8f8b-d714ea22a6b2.png)



> (It doesn't seem right to say "has documentation" because often changes just update docs instead of adding it.)

i think it it alright because even if docs are updated the pr still has docs

> How do we deal with semantically redundant labels? If we changed "documentation" to "8.has: documentation", this will break existing queries, and we'd have to update contributing guides for example.

the documentation label is for issues (nixpkgs has [`9.needs: documentation`](https://github.com/NixOS/nixpkgs/labels?q=documentation))  while ` 8.has: documentation` is for PRs

Reply:
@Artturin thanks for the input. Is that documented anywhere else? I think part of this PR should be at least a short blurb in the [maintainers' handbook](https://github.com/NixOS/nix/tree/master/maintainers) or [contribution guide](https://github.com/NixOS/nix/tree/master/doc/manual/src/contributing).

On the [last team meeting](https://discourse.nixos.org/t/2022-12-12-nix-team-meeting-minutes-16/24119/1) we decided not to adopt the new labeling scheme yet, and the general sentiment is to keep the process overhead low. I think we should take another look at this given the new information.

Reply:
i found the email thread from 2015 in which the label system was created by @nbp https://releases.nixos.org/nix-dev/2015-November/018714.html